Cython/Compiler/ExprNodes.py

```diff
@@ -263,6 +263,20 @@ def translate_cpp_exception(code, pos, inside, py_result, exception_value, nogil
     code.putln(code.error_goto(pos))
     code.putln("}")
 
+def needs_cpp_exception_conversion(node):
+    # This is only called after confirming that node.exception_check == "+"
```

Comment:
If this is worth a comment, it seems worth an assertion.
```suggestion
    assert node.exception_check == "+"
```